cicd/README.md

```diff
@@ -0,0 +1,55 @@
+# Javabuilder CI/CD
+
+Objectives:
+
+* General
+  * Provide fast feedback on Pull Requests of any build/verify failures
+  * Run all integration/acceptance tests in a non-prod environment
+  * Provide a non-prod development environment that local resources can interact with
+  * Easily create an ad-hoc environment for a given branch, which updates with changes
+  * Eliminate or minimize manual AWS Console configuration
+  * Use policies and roles with least-privilege
+* **CodeBuild** For every PR created that merges into `main`
+  * Build the project, ensure it compiles/transpiles/validates
+  * Run all tests
+  * Update the PR with the build/test status
+  * Provide easily accessible logs and artifacts from the build/test job
+* **CodePipeline** For every commit merged to `main`
+  * Run tests and build artifacts necessary for deployment
+  * Deploy the application to a Test environment
+    * As prod-like as possible
+  * Run integration and acceptance tests in the Test environment
+  * Make test results and artifacts easily accessible to developers
+  * If Test environment passes all tests, then deploy to Prod environment
+  * Deploy the application to a Dev environment
```

Comment:
This last line doesn't seem like it fits with the other. Is this a separate dev environment similar to the ones we deploy to test out AWS changes or is this something similar to the staging environment?

Reply:
The idea here would be an environment that developers can interact with and not worry about impacting production or the cicd pipeline. Resources might be running at a much lower scale, and access to these resources via the AWS console would be more permissive to support troubleshooting and issue replication.

IMO the Test environment should be as prod-like as possible and protected from out-of-pipeline changes. TBD whether a third environment running `main` is actually useful to the team.

I've usually had this environment get updated first, and a successful deployment to Dev be a prerequisite for deploying to Test - but I could also see this being a parallel deployment.

Reply:
Ahh ok. That makes sense.